cmd/pj-rehearse/main.go

```diff
@@ -306,7 +318,11 @@ func rehearseMain() error {
 
 	resolver := registry.NewResolver(refs, chains, workflows, observers)
 	jobConfigurer := rehearse.NewJobConfigurer(prConfig.CiOperator, resolver, prNumber, loggers, rehearsalTemplates.Names, rehearsalClusterProfiles.Names, jobSpec.Refs)
```

Comment:
Note that `prConfig` also has `Prow` member, so you do not need to load it separately by `loadProwConfig` and therefore you don't even need that method. The convention in `pj-rehearse` is that we have complicated structs (that carry the "context" of the rehearsals)  that have simpler methods (that operate moslty over just jobs to be rehearsed).

So ideally `NewJobConfigurer` should either receive `prConfig.Prow` (I assume you want to use the PR version of the config, not the baseline) and save it into the struct, and `ConfigurePresubmitRehearsals` should continue to just take the jobs as an input.